scm-core/src/main/java/sonia/scm/repository/work/WorkdirProvider.java

```diff
@@ -24,30 +24,45 @@
 
 package sonia.scm.repository.work;
 
+import sonia.scm.repository.RepositoryLocationResolver;
+
+import javax.inject.Inject;
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
+import java.nio.file.Path;
 
 public class WorkdirProvider {
 
   private final File rootDirectory;
+  private final RepositoryLocationResolver repositoryLocationResolver;
 
-  public WorkdirProvider() {
-    this(new File(System.getProperty("scm.workdir" , System.getProperty("java.io.tmpdir")), "scm-work"));
+  @Inject
+  public WorkdirProvider(RepositoryLocationResolver repositoryLocationResolver) {
+    this(new File(System.getProperty("scm.workdir" , System.getProperty("java.io.tmpdir")), "scm-work"), repositoryLocationResolver);
   }
 
-  public WorkdirProvider(File rootDirectory) {
+  public WorkdirProvider(File rootDirectory, RepositoryLocationResolver repositoryLocationResolver) {
     this.rootDirectory = rootDirectory;
+    this.repositoryLocationResolver = repositoryLocationResolver;
     if (!rootDirectory.exists() && !rootDirectory.mkdirs()) {
       throw new IllegalStateException("could not create pool directory " + rootDirectory);
     }
   }
 
   public File createNewWorkdir() {
+    return createWorkDir(this.rootDirectory);
+  }
+
+  public File createNewWorkdir(String repositoryId) {
+    return createWorkDir(repositoryLocationResolver.forClass(Path.class).getLocation(repositoryId).toFile());
```

Comment:
I think we should not create the work directories in the root of the repository directory. We should add a stable parent like `workdir` and then create the temp directories as child of it.